### DISCUSSION TITLE
Migration of old DAW project to CLAP

### DISCUSSION THREAD

**Discussion by dwuertz** [free-audio/clap#211] [Ideas]
When a plugin vendor introduces CLAP support for his products which are already on the market, users might want to migrate their old projects that are still using VST/AU/whatever versions.

In the VST3 SDK, there exists a concept to tell the host application that a VST3 with a given UID may replace a VST2 plugin with a certain ID.

https://steinbergmedia.github.io/vst3_dev_portal/pages/Tutorials/Guideline+for+VST3+replacing+VST2.html

Has a similar idea been discussed for CLAP previously (i.e. allow VST2/VST3/AAX/AU -> CLAP migration)? My guess is that could be handled by some extension API.

**Comment by defiantnerd** [comment]
There are two ways to migrate VST3(etc.) to CLAP:

You can use the [clap-wrapper project](https://github.com/free-audio/clap-wrapper) and specifically the `clap_plugin_info_as_vst3` structure as in https://github.com/free-audio/clap-wrapper/blob/main/include/clapwrapper/vst3.h. This way, a small wrapper VST3 can be installed that actually runs the CLAP in VST3 hosts. This is already possible and works with all hosts. 

The second option would be an - yet to define - extension to allow the host directly to swap the VST3 (or others) against the CLAP version. Yet, I am not sure if this really should be direct part of the CLAP SDK itself and if this is a viable solution for developers. And it only works when the host would natively support CLAP.
